src/util/address.rs

```diff
@@ -222,48 +363,88 @@ impl Display for Address {
                 };
                 prefixed[1..].copy_from_slice(&hash[..]);
                 base58::check_encode_slice_to_fmt(fmt, &prefixed[..])
-            },
-            Payload::WitnessProgram(ref witprog) => {
-                fmt.write_str(&witprog.to_address())
-            },
+            }
+            Payload::WitnessProgram {
+                version: ver,
+                program: ref prog,
+            } => {
+                let mut b32_data = vec![ver];
+                b32_data.extend_from_slice(&prog.to_base32());
```

Comment:
I think this could be done without the extra allocation by using `ToBase32::write_base32`.

Reply:
Cool, done!